File: examples/Spark/simpleapp/src/main/java/SimpleApp.java
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class SimpleApp {
  public static void main(String[] args) {
    SparkConf conf = new SparkConf().setAppName("SimpleApp").setMaster("local");
    JavaSparkContext sc = new JavaSparkContext(conf);

    String logFile = "/home/amit/spark-install/spark/README.md"; //Update to your file path 
    JavaRDD<String> logData = sc.textFile(logFile);

    long numAs = logData.filter(s -> s.contains("a")).count();
    long numBs = logData.filter(s -> s.contains("b")).count();
    
    System.out.println();
    System.out.println("Lines with a: " + numAs + ", lines with b: " + numBs);
    System.out.println();

    
    JavaRDD<Integer> lineLengths = logData.map(s -> s.length());
    int totalLength = lineLengths.reduce((a, b) -> a + b);
    
    System.out.println();
    System.out.println("Total linelengths = " + totalLength);
    System.out.println();


    sc.stop(); // stop the Spark context
    sc.close();
  }
}

File: examples/spark/simpleapp-python/SimpleApp.py
#!/usr/bin/env python3
"""SimpleApp.py"""
from pyspark.sql import SparkSession

spark = SparkSession.builder.appName("SimpleApp").getOrCreate()

logFile = "../simpleapp/README.md"  # Should be some file on your system
logData = spark.read.text(logFile).cache()

numAs = logData.filter(logData.value.contains('a')).count()
numBs = logData.filter(logData.value.contains('b')).count()

print("Lines with a: %i, lines with b: %i" % (numAs, numBs))

spark.stop()
